dts/arm/nxp/nxp_lpc55Sxx_clocks.dtsi

```diff
@@ -0,0 +1,888 @@
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		ostimer32khzclk: ostimer32khzclk@4002009c {
+			compatible = "nxp,syscon-clock-gate";
+			#clock-cells = <1>;
+			/* PMC::OSTIMER[CLOCKENABLE] */
+			reg = <0x4002009c 0x1>;
+			offset = <0x1>;
+
+			ostimer32khz_clock: ostimer32khz-clock {
+				compatible = "clock-output";
+				#clock-cells = <0>;
+			};
+		};
+
+		osc32khz_clock: osc32khz-clock {
+			compatible = "clock-output";
+			#clock-cells = <0>;
+		};
+
+		rtcclk1hzdiv: rtcclk1hzdiv@40020098 {
+			compatible = "nxp,syscon-rtcclk";
+			#clock-cells = <1>;
+			/* PMC::RTCOSC32K[CLK1HZDIV] */
+			reg = <0x40020098 0xb>;
+			offset = <0x10>;
+			add-factor = <31744>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			rtc_1hz_clk: rtc-1hz-clk@4002c000 {
+				compatible = "nxp,syscon-clock-gate";
+				#clock-cells = <1>;
+				/* RTC::CTRL[RTC_EN] */
+				reg = <0x4002c000 0x1>;
+				offset = <0x7>;
+
+				rtc1hz_clock: rtc1hz-clock {
+					compatible = "clock-output";
+					#clock-cells = <0>;
+				};
+			};
+		};
+
+		rtcclk1khzdiv: rtcclk1khzdiv@40020098 {
+			compatible = "nxp,syscon-rtcclk";
+			#clock-cells = <1>;
+			/* PMC::RTCOSC32K[CLK1KHZDIV] */
+			reg = <0x40020098 0x3>;
+			offset = <0x1>;
+			add-factor = <28>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			rtc_1khz_clk: rtc-1khz-clk@4002c000 {
+				compatible = "nxp,syscon-clock-gate";
+				#clock-cells = <1>;
+				/* RTC::CTRL[RTC1KHZ_EN] */
+				reg = <0x4002c000 0x1>;
+				offset = <0x6>;
+
+				rtc1khz_clock: rtc1khz-clock {
+					compatible = "clock-output";
+					#clock-cells = <0>;
+				};
+			};
+		};
+	};
+
+	pll0clksel: pll0clksel@40000290 {
+		compatible = "nxp,syscon-clock-mux";
+		#clock-cells = <1>;
+		/* SYSCON::PLL0CLKSEL[SEL] */
+		reg = <0x40000290 0x3>;
+		offset = <0x0>;
+		input-sources = <&fro_12m &clk_in_en &fro_1m &rtcosc32ksel
+				&no_clock>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		pll0: pll0@40000580 {
+			compatible = "nxp,lpc55sxx-pll0";
+			reg = <0x40000580 0x20>;
+			#clock-cells = <9>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			pll0_pdec: pll0-pdec@4000058c {
+				compatible = "nxp,lpc55sxx-pll-pdec";
+				#clock-cells = <1>;
+				/* SYSCON::PLL0PDEC[PDIV] */
+				reg = <0x4000058c 0x5>;
+			};
+		};
+	};
+
+	pll0_directo: pll0-directo@40000580 {
+		compatible = "nxp,syscon-clock-mux";
+		#clock-cells = <1>;
+		/* SYSCON::PLL0CTRL[BYPASSPOSTDIV] */
+		reg = <0x40000580 0x1>;
+		offset = <0x14>;
+		input-sources = <&pll0_pdec &pll0>;
+	};
+
+	pll0_bypass: pll0-bypass@40000580 {
+		compatible = "nxp,syscon-clock-mux";
+		#clock-cells = <1>;
+		/* SYSCON::PLL0CTRL[BYPASSPLL] */
+		reg = <0x40000580 0x1>;
+		offset = <0xf>;
+		input-sources = <&pll0_directo &pll0clksel>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		pll0div: pll0div@400003c4 {
+			compatible = "nxp,syscon-clock-div";
+			#clock-cells = <1>;
+			/* SYSCON::PLL0CLKDIV[DIV] */
+			reg = <0x400003c4 0x8>;
+		};
+	};
```

Comment:
I'm looking your work and it is very interesting. I liked how the `rtcosc32ksel` node was created. However, it is a little more difficult to understand the `pll0`. I didn't see the `compatible = "clock-output";` for instance. Then, how it is arranged is hard to understand what will be the outputs, I imagine that is because of the nature of the PLL itself. So, the `clock-output-names` can be omitted because now there is the `clock-output`, is this correct?

In this case, a PLL with multiple outputs will be something like:

```c
pll {
  my-out-0 {
    compatible = "clock-output";
  };
  my-out-1 {
    compatible = "clock-output";
  };
  ...
  my-out-n {
    compatible = "clock-output";
  };
};
```

I was wondering if you could elaborate to us a little bit more about it.